baremetal/metal3machine_manager.go

```diff
@@ -1338,15 +1335,58 @@ func (m *MachineManager) SetNodeProviderID(ctx context.Context, bmhID, providerI
 		m.Log.Info("Target node is not found, requeuing")
 		return &RequeueAfterError{RequeueAfter: requeueAfter}
 	}
+	namespace := m.Metal3Machine.GetNamespace()
+	m3mName := m.Metal3Machine.GetName()
+	bmhName, err := m.getBmhNameFromM3Machine()
+	if err != nil {
+		m.Log.Info("unable to retrieve BMH from Metal3Cluster")
+		return &RequeueAfterError{RequeueAfter: requeueAfter}
+	}
+	providerIDLegacy := *providerID
+	providerIDNew := fmt.Sprintf("metal3://%s/%s/%s", namespace, bmhName, m3mName)
+	if !m.Metal3Cluster.Spec.NoCloudProvider {
```

Comment:
you could separate the content of this if into another function. It would be more readable

Reply:
thanks, will do that.

Reply:
Here the results are evaluated. The actual content has moved to the end of the file